sweepbatcher/sweep_batch.go

```diff
@@ -543,10 +548,25 @@ func (b *batch) Run(ctx context.Context) error {
 		}
 	}
 
+	// skipBefore is the time before which we skip batch publishing.
+	// This is needed to facilitate better grouping of sweeps.
+	// For batches loaded from DB waitingPeriod should be 0.
+	skipBefore := time.Now().Add(b.cfg.waitingPeriod)
```

Comment:
nit: instead of using real time, we could adopt the `lnc/clock` package so we can simulate time better in tests as needed.

Reply:
Done. I added this in a separate commit "sweepbatcher: use lnd/clock for timers".

While updating the test, I found that the batch starts empty and then the first sweep is added. This results in the order of RegisterSpendNtfn and timer registrations being nondeterministic. I decided to catch this in the test via two separate goroutines to avoid deadlocks. Maybe the batch can be reworked in the future make sure it always calls RegisterSpendNtfn before starting any timers, then the code of the test can be simplified.

Testing for urgent batch failed sometimes, because the check happen to run before addSweep and the batch was empty at that point (timeout unknown). To fix this, the check was moved to timerChan handler, just before the code that checks if waiting period has ended, before publishing. By that time the sweep is added.

Also I found that lnd/clock is imported and stored as a field in store.go, but not used. I removed it in a separate commit.

Reply:
> Maybe the batch can be reworked in the future make sure it always calls RegisterSpendNtfn before starting any timers, then the code of the test can be simplified.

Agreed!